index.html

```diff
@@ -1727,12 +1727,14 @@ <h3>Issuer</h3>
         <dl>
           <dt><var id="defn-issuer">issuer</var></dt>
           <dd>
-The value of the <code>issuer</code> <a>property</a> MUST be either a
-<a>URL</a> or an object containing an <code>id</code> <a>property</a>. It is
-RECOMMENDED that the <a>URL</a> in the <code>issuer</code> or its
-<code>id</code> be one which, if dereferenced, results in a document containing
-machine-readable information about the <a>issuer</a> that can be used to
-<a>verify</a> the information expressed in the <a>credential</a>.
+The value of the <code>issuer</code> <a>property</a> MUST either be a
+<a>URL</a>, or be an object containing an <code>id</code> <a>property</a>
```

Comment:
So if you encode the issuer as `{"@id": "the:url"}`, that's not a valid VC? Or if your `@context` establishes another shorthand for `@id`, it's not conforming to use that? How would a pure JSON-LD processor detect those issues: do they need to check the incoming representation before expanding it to RDF?

Reply:
VCs are a subset of JSON-LD documents. Every VC is a valid JSON-LD document, but not every JSON-LD document is a valid VC.

> So if you encode the issuer as {"@id": "the:url"}, that's not a valid VC? 

Correct, it is not, because [the Verifiable Credentials specification requires that JSON-LD compact document form is used](https://w3c.github.io/vc-data-model/#json-ld).

> Or if your @context establishes another shorthand for @id, it's not conforming to use that?

No, it wouldn't be a valid VC because it wouldn't follow the data model provided in the specification. It would, however, be a valid JSON-LD document and the semantics wouldn't be any different from a valid VC.

> How would a pure JSON-LD processor detect those issues: do they need to check the incoming representation before expanding it to RDF?

Yeah, most every implementation that we know of uses JSON Schema, or hard-coded checks that are roughly equivalent to JSON Schema, to check the incoming form for the VCs they expect to process.

Reply:
Ok, thanks, and sorry for assuming a different model for these constraints than you actually had. I'll update my understanding.

Reply:
No apologies necessary... it's important to understand how folks are interpreting the mental model for VCs differently. One of the things we're finding is that people that go beyond "I'll just copy/paste an example" (which is the mental model that made schema.org so successful) into actually trying to reconcile the subset of JSON-LD that we're using here tend to get tripped up in areas such as this.

In general, the approach we're taking is to optimize for the average developer that's going to be copy/pasting examples into their code/systems. We're certainly not there yet wrt. ecosytem maturity. Developers still have to think too much when using VCs today, but we're hoping that a schema.org-like approach in different market verticals will help here.

We have some of the constraints in the VC spec that we do because we want people to be able to process these objects just like they would process any other JSON object (after they've verified the digital signature). IOW, in the general case, we don't want to force them to understand JSON-LD intricacies or go to RDF to start using the data. Many of the implementations today only break out the JSON-LD processors and go to RDF when they verify the digital signature -- once that's done, they just work w/ the VC as they would most other JSON input into their systems.